mep/accounts/partial_date.py

```diff
@@ -67,6 +67,11 @@ def date_format(value):
         '''Return a format string for use with :meth:`datetime.date.strftime`
         to output a date with the appropriate precision'''
         parts = []
+
+        # Handle NULL as indicating full date precision
```

Comment:
This addition conflicts with the `not value` check below which is commented as "no precision = no date"

What was the scenario that required this addition?

Reply:
@rlskoeser It seems like the notes there were slightly out of sync. `None` isn't something we ever actually set on the partial dates as currently implemented. The only time `None` comes up now is when we have dates there were previously not partial dates -- and thus need to be treated as full precision.

Reply:
I was a little puzzled about this because the default value for them would be `None` on `Borrrow` and `Purchase` objects. But if the only way we ever set them was through the admin form or the partial date descriptor, then it will always set the precision along with the date. In that case, the only time we would ever get a `None` value would be when the date is not set, in which case the `date_format` method never gets called.

Following up on this for two reasons:
- Your addition of handling `None` as full precision does seem like what we need for events where we're not using partial dates; how do we reconcile with later block (`if not value`) or clarify in the comments why they are different? If we hit that block, does it mean it is explicitly set to no precision? (not sure that's a meaningful case but I guess it is possible)
- I'm no longer certain that we should set the default to full precision #341 - does it make any sense if there is no date set?  I experimented a little with setting the default as full precision and can see some of the changes you talked about that need to be made for it to work, but I think conceptually this is the sticking point. 

Not a blocker for merging this PR, but would like your opinion on #341 and think we need to clarify the comments for the two unset cases in `date_format`.

Reply:
I think the full precision as default on #341 is undesirable. `None` | `Null` or full precision equally full precision is accurate and easy to write for if necessary using a `Q` statement.

I've tried to revise the comments in `date_format` to reflect reality. One is a strict check for `None`, whereas the other is looking for an empty `DateFormat`, i.e. `0`.